pkgs/development/python-modules/conduit/default.nix

```
@@ -0,0 +1,72 @@
+{
+  lib,
+  pkgs,
+  buildPythonPackage,
+  setuptools,
+  numpy,
+  pip,
+  openmpi,
+
+  mpiSupport ? false,
+}:
+let
+  conduit = pkgs.conduit.override { inherit mpiSupport; };
+
+  # As cmake is invoked by setup.py, adding openmpi to buildInputs is not enough for it to be found.
+  # It is necessary to explicitly provide openmpi's location through cmake flags.
+  mpi = lib.getDev openmpi;
+  mpiCmakeFlags = lib.concatStringsSep "', '" [
+    (lib.cmakeFeature "MPI_C_COMPILER" (lib.getExe' mpi "mpicc"))
+    (lib.cmakeFeature "MPI_CXX_COMPILER" (lib.getExe' mpi "mpicxx"))
+  ];
+in
+buildPythonPackage {
+  inherit (conduit)
+    pname
+    version
+    src
+    nativeBuildInputs
+    buildInputs
+    ;
+  pyproject = true;
+
+  postPatch =
+    ''
+      substituteInPlace setup.py \
+        --replace-fail \
+          "'-j2'" \
+          "f'-j{os.environ.get(\"NIX_BUILD_CORES\")}'"
+    ''
+    + lib.optionalString mpiSupport ''
+      substituteInPlace setup.py \
+        --replace-fail \
+          "cmake_args = [" \
+          "cmake_args = ['${mpiCmakeFlags}', "
```

What about?

```suggestion
          "cmake_args = ['$cmakeFlags', "
```

(Obviously this won't work verbatim, but the general idea)

---

Ah I see the problem, cmakeFlags is only available within the cmake configure hook, anyway it's not even necessary for finding mpi.